app.rb

```diff
@@ -0,0 +1,161 @@
+require_relative 'book'
+require_relative 'person'
+require_relative 'rental'
+require_relative 'student'
+require_relative 'teacher'
+
+class App
+  MENU_OPTIONS = {
+    '1' => :list_books,
+    '2' => :list_people,
+    '3' => :create_person,
+    '4' => :create_book,
+    '5' => :create_rental,
+    '6' => :list_all_rentals,
+    '7' => :exit_program
+  }.freeze
+
+  def initialize
+    @books = []
+    @rentals = []
+    @people = []
+  end
+
+  def start
+    loop { choose_option }
+  end
+
+  def list_books
+    if @books.empty?
+      puts 'No books available. Please add some books and try again.'
+    else
+      puts "There are #{@books.count} book(s) available."
+      @books.each_with_index do |book, index|
+        puts "#{index + 1}) Title: \"#{book.title}\" | Author: #{book.author}"
+      end
+    end
+  end
+
+  def list_people
+    if @people.empty?
+      puts 'No people added yet. Please add a person and try again.'
+    else
+      @people.each_with_index do |person, index|
+        puts "#{index + 1}) [#{person.class}] Name: #{person.name} | Age: #{person.age} | ID: #{person.id}"
+      end
+    end
+  end
+
+  def create_person
+    print 'Would you like to create a student (1) or a teacher (2)? Select a number: '
+    choice = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    case choice
+    when '1'
+      create_student(name, age)
+    when '2'
+      create_teacher(name, age)
+    else
+      puts 'Invalid choice. Please try again.'
+    end
+  end
+
+  def create_student(name, age)
+    print 'Grade: '
+    grade = gets.chomp
+    print 'Parent permission? [Y/N]: '
+    permission_option = gets.chomp.downcase
+    student = Student.new(grade, name, age, parent_permission: permission_option == 'y')
+    @people << student
+    puts "Student created successfully. ID is #{student.id}"
+  end
+
+  def create_teacher(name, age)
+    print 'Specialization: '
+    specialization = gets.chomp
+    teacher = Teacher.new(specialization, name, age)
+    @people << teacher
+    puts "Teacher created successfully. Teacher ID is #{teacher.id}"
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+
+    if title.strip != '' && author.strip != ''
+      book = Book.new(title, author)
+      @books << book
+      puts 'Book created successfully.'
+    else
+      puts 'Please enter the book title and author.'
+    end
+  end
+
+  def create_rental
+    if @books.empty? || @people.empty?
+      puts 'Nothing to see here.'
+      return
+    end
+
+    puts 'Enter the number of the book you want: '
+    list_books
+    book_number = gets.chomp.to_i - 1
+
+    puts 'Enter your ID: '
+    list_people
+    person_id = gets.chomp.to_i
+    individual = @people.find { |person| person.id == person_id }
```

Comment:
- [ ] While your project works right now, kindly note that the previous reviewer @KanzaTahreem  was right [here](https://github.com/axcell-0/school-library/pull/4#discussion_r1366577881) since you are not supposed to get the person by their ID but rather via their index (number). Please refer to [this short video](https://youtu.be/vkkgrhD6aXQ?si=DmXaL0jdmrwIW1IF) provided in the project requirements. **Please fast forward to around minute 1:20 there**.

However, that is a very easy fix to make use of the index and not the user ID I will add the line for you. Replace the current  individual variable with this one: 

```
individual = @people[person_id - 1]
```

Above we are basically using the @people array and its indexing to access the person that corresponds to the given person_id. 

**Do not forget to update the text above telling the user to use the index and not the ID.**

**Just to show you that it works, below is a screencast:**

[Screencast from 20-10-2023 11:28:47 ASUBUHI.webm](https://github.com/axcell-0/school-library/assets/86318284/ec9af5f5-400c-4a48-9119-cf58b47d72fe)